src/AppBundle/Controller/DefaultController.php

```diff
@@ -34,4 +36,59 @@ public function loginAction(Request $request)
             'form'  => $form->createView(),
         ]);
     }
+
+    /**
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function swaggerUiAction()
+    {
+        $docUrl = $this->get('service_container')->getParameter('app_swagger_ui_resource_url');
+
+        if (preg_match('/^(https?:)?\/\//', $docUrl)) {
+            // If https://..., http://..., or //...
+            $url = $docUrl;
+        } elseif (strpos($docUrl, '/') === 0) {
+            //If starts with "/", interpret as an asset.
+            $url = $this->get('templating.helper.assets')->getUrl($docUrl);
+        } else {
+            // else, interpret as route-name.
+            $url = $this->generateUrl($docUrl, [
+                'resource' => $this->get('service_container')->getParameter('app_swagger_ui_static_resource_filename')
+            ]);
+        }
+
+        $url = rtrim($url, '/');
+
+        return $this->render(':default:swagger_ui.html.twig', [
+            'resource_url' => $url
+        ]);
+    }
+
+    /**
+     * @param string $resource
+     * @return JsonResponse
+     */
+    public function swaggerUiStaticResourceAction($resource)
+    {
+        $dir = sprintf(
+            '%s/../%s',
+            $this->get('kernel')->getRootDir(),
+            $this->get('service_container')->getParameter('app_swagger_ui_static_resource_dir')
+        );
+
+        try {
+            $finder = new Finder();
+            $files = $finder->in($dir)->files()->name($resource);
+
+            if (count($files) === 0) {
+                throw new \Exception(sprintf('Cannot find API Documentation: %s', $resource));
+            }
+
+            $doc = file_get_contents($dir . $resource);
+
+            return new JsonResponse(json_decode($doc));
+        } catch (\Exception $e) {
+            throw $this->createNotFoundException($e->getMessage());
```

Comment:
The third argument of any exception is a previous exception. So pass previous exception as third argument and write some understandable exceptions by hands